scripts/mac_export_load.sh

```diff
@@ -0,0 +1,159 @@
+
+
+# This script EXPECTS to be executed from the ROOT of a buildable mac project	
+
+# REQUIREMENTS :
+# Java Development Kit 11+ (JRE is not enough as we use the jps command)
+# Maven
+# ActiveViam artifacts access
+# Valid ActivePivot License
+# cURL
+# jq
+
+# INPUT : 
+# - 1 String : string containing the exported folder string
+# - 2 Optional String : Path to the maven settings file, this file is expected to grant read access to ActiveViam artifacs
+
+# OUTPUT : 
+# - File : Server logs
+# The server logs of both sandbox and mac applications as well as the maven outputs will be saved in the
+# "logs" subfolder of the execution directory of the script
+
+
+# Functions are defined to avoid command duplication and make the main script easily readable
+# Make sure any variable used are defined before calling these
+check_root(){
+	#Just making sure the execution path is at the Root of a mac project
+	if [[ -z $(cat pom.xml) || -z $(cat pom.xml | grep "<artifactId>${MAC_ARTIFACTID}</artifactId>") ]]; then
+		echo "The execution directory of the script ${PWD} is not the root of a buildable mac project."
+	exit 1
+	fi	
+}
+
+check_requirements(){
+	#Java
+	#No version check as it's too vendor-dependent on the installed JDK
+	if [ -z $(which java) ]; then
+		echo "Java is not installed."
+		exit 1
+	#Maven
+	elif [ -z $(which mvn) ]; then
+		echo "Maven is not installed."
+		exit 1
+	#cURL
+	elif [ -z $(which curl) ]; then
+		echo "cURL is not installed."
+		exit 1
+	#jq
+	elif [ -z $(which jq) ]; then
+		echo "jq is not installed."
+		exit 1
+	fi
+}
+
+cleanup(){
+	echo "Killing the java processes and removing temporary files..."
+	kill -n 15 ${PID_MAC}
+	rm -rdf ${BASE_DIR}/queries/output
+}
+
+# Function executing a remote call to the a loaded MAC cube available at localhost:9092
+# Usage : check_query queryId
+# queryId : String , Id of the checked query, its payload is expected to be available at ${BASE_DIR}/queries/input/${queryId}.json
+#           and the values obtained for the cells to match the content of ${BASE_DIR}/queries/ref/${queryId}.txt
+check_query(){
+	if [ -z "$1" ];	then
+		echo "No argument supplied to the query function"
+		exit 1
+    else
+    	curl -X POST -u admin:admin -H "Content-Type:application/json" \
+    		-d @${BASE_DIR}/queries/input/$1.json \
+    		http://localhost:9092/activeviam/pivot/rest/v8/cube/query/mdx \
+    		| jq .cells[].value > ${BASE_DIR}/queries/output/$1.txt
+    fi
+
+    if [ -z $(diff --strip-trailing-cr --ignore-all-space ${BASE_DIR}/queries/output/$1.txt ${BASE_DIR}/queries/ref/$1.txt) ]; then
+    	echo $1"... OK"
+    else
+    	echo "Error when comparing expected query output to query result:"
+    	echo $(diff --strip-trailing-cr --ignore-all-space ${BASE_DIR}/queries/output/$1.txt ${BASE_DIR}/queries/ref/$1.txt)
+	     # Do the cleanup to make sure we don't leave open processes
+	    cleanup
+	    exit 1
+	fi
+}
+
+###################
+# MAIN SCRIPT START
+
+if [ -z "$1" ]; then
```

Comment:
can we put this part in a method e.g. parse_args() ? Could be nice to have a help section if $1 is:
--help | -h | -?
print_help()
exit 0

Reply:
As said earlier, it's doable but I don't see the point, since that script will never realistically be called with that argument